src/routes/v1/secret_test.go

```diff
@@ -11,76 +12,376 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestCreateSecret(t *testing.T) {
-	secretRequest := types.CreateSecretRequest{
-		Type:       "opaque",
-		SecretName: "new-secret",
-		Data:       []types.KeyValue{{Key: "key1", Value: "ZmFrZQ=="}},
-	}
-	body, _ := json.Marshal(secretRequest)
-	request, _ := http.NewRequest("POST", "/v1/namespaces/default/secrets/", bytes.NewBuffer(body))
-	request.Header.Set("Content-Type", "application/json")
-
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.CreateSecretResponse
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
-	assert.Equal(t, "new-secret", response.SecretName)
-	assert.Equal(t, "default", response.NamespaceName)
+func TestGetSecret(t *testing.T) {
+	secretNamespace := "test-namespace-secret"
+	secretName := "test-secret"
+	type requestUri struct {
+		namespace string
+		secret    string
+	}
+
+	type want struct {
+		statusCode int
+		response   map[string]interface{}
+	}
+
+	cases := map[string]struct {
+		requestUri requestUri
+		want       want
+	}{
+		"ShouldSuccessGettingSecret": {
+			requestUri: requestUri{
+				secret:    secretName,
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: map[string]interface{}{
+					"secretName": secretName,
+					"type":       "Opaque",
+					"id":         "",
+					"data":       []interface{}{map[string]interface{}{"key": "key1", "value": "fake"}},
+				},
+			},
+		},
+		"ShouldNotFoundSecret": {
+			requestUri: requestUri{
+				secret:    "test-not-exists",
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					"details": fmt.Sprintf("secrets %q not found", "test-not-exists"),
+					"error":   "Operation failed",
+				},
+			},
+		},
+		"ShouldNotFoundNamespace": {
+			requestUri: requestUri{
+				secret:    secretName,
+				namespace: "test-not-exists",
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+				response: map[string]interface{}{
+					"details": fmt.Sprintf("secrets %q not found", secretName),
+					"error":   "Operation failed",
+				},
+			},
+		},
+	}
+	createTestNamespace(secretNamespace)
+	createTestSecret(secretName, secretNamespace)
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			request, _ := http.NewRequest("GET", fmt.Sprintf("/v1/namespaces/%s/secrets/%s",
+				test.requestUri.namespace, test.requestUri.secret), nil)
```

Comment:
Modify the requests to look like this:

https://github.com/dana-team/platform-backend/blob/1989d979499464dac34fd2dbb08b0ea01c502f3a/src/routes/v1/capp_test.go#L314

Also use `http.MethodGet` instead of `GET` and check for the error in `http.NewRequest`